### PR TITLE
Fix Team Color Inconsistencies

### PR DESCRIPTION
**Pull request by tahmid-23**

### CODE REVIEW

File: patches/api/0007-Adventure.patch [hunk 1]
```diff
@@ -3801,27 +3826,30 @@ index 0db7fe1b9fe5621ceed3f4f046691e359f5949dd..30fce0df75494eb9b7409f08ea3d6ff8
...
       * @param color new color, must be non-null. Use {@link ChatColor#RESET} for
       * no color
-+     * @deprecated in favour of {@link #color(net.kyori.adventure.text.format.NamedTextColor)}
++     * @throws IllegalStateException if this team has been unregistered
++     * @deprecated in favour of {@link #color(net.kyori.adventure.text.format.NamedTextColor)} or {@link #decoration(net.kyori.adventure.text.format.TextDecoration)}
       */
+-    void setColor(@NotNull ChatColor color);
 +    @Deprecated // Paper
-     void setColor(@NotNull ChatColor color);
++    void setColor(@NotNull ChatColor color) throws IllegalStateException; // Paper - add IllegalStateException to throws
```
Comment (kennytv) [inline patches/api/0007-Adventure.patch:3849]: This is unnecessary

**Review by kennytv — CHANGES_REQUESTED** [review]
We can't just randomly break nullability in api

**Comment by tahmid-23** [comment]
> We can't just randomly break nullability in api

sorry, I didn't know that was a breaking change.
should I keep the existing behavior?

File: patches/api/0007-Adventure.patch (hunk 1 above)
Comment (tahmid-23) [reply patches/api/0007-Adventure.patch:3849]: do I keep the @throws since it can throw IllegalStateException or remove it all entirely

**Comment by kennytv** [comment]
Keep existing behavior and just only keep a throws tag in the javadoc.

But generally, "teamcolor" isn't just called color for funzies in Vanilla, but because the game doesn't actually fully allow/support decorations there.

**Comment by tahmid-23** [comment]
> But generally, "teamcolor" isn't just called color for funzies in Vanilla, but because the game doesn't actually fully allow/support decorations there.
```java
            Scoreboard scoreboard = Bukkit.getScoreboardManager().getNewScoreboard();
            Team team = scoreboard.registerNewTeam("team");
            team.displayName(Component.text("hello"));
            team.setColor(ChatColor.UNDERLINE);
            team.addPlayer(event.getPlayer());
            event.getPlayer().setScoreboard(scoreboard);
```
granted I am on forge but this changed my name's appearance in the tablist for the decorations I did test
don't know if vanilla does it but if at least forge does it then I think it's worth having adventure equivalents

**Comment by kezz** [comment]
See https://github.com/PaperMC/Paper/issues/5302

**Comment by tahmid-23** [comment]
> See #5302

hm okay I will test again and then close if not so

**Comment by kezz** [comment]
The point is that the current behaviour has been marked as working as intended, regardless of whether or not the client supports such values being sent in the packet.

**Comment by kennytv** [comment]
Adding to the linked issue, Vanilla doesn't allow this in any meaningful way and even prevents the decoration from being saved on a restart for permanent scoreboards. So adding API for that here would be a bit volatile, sorry.

**Pull request closed without merging**